librosa/feature/spectral.py

```diff
@@ -705,7 +720,10 @@ def spectral_rolloff(
 
     # Make sure that frequency can be broadcast
     if freq.ndim == 1:
-        freq = freq.reshape((-1, 1))
+        # reshape for broadcasting
+        shape = [1 for _ in range(S.ndim)]
+        shape[-2] = -1
+        freq = freq.reshape(shape)
 
     total_energy = np.cumsum(S, axis=0)
```

Comment:
this sum is probably on the wrong axis

Reply:
Re-upping this: i think this cumsum should be on `axis=-2`

Reply:
similarly, the usage of `total_energy` below should probably move to `total_energy[..., -1, :]`, and the nanmin axis should be -2.